src/story.ts

```diff
@@ -106,37 +110,33 @@ export type ArgTypesEnhancer<TRenderer extends Renderer = Renderer, TArgs = Args
   secondPass?: boolean;
 };
 
-export type StoryContextUpdate<TArgs = Args> = {
+export interface StoryContextUpdate<TArgs = Args> {
   args?: TArgs;
   globals?: Globals;
   // NOTE: it is currently possibly to add *any* key you like to the context
   // (although you cannot override the basic keys). This will likely be removed in future.
   [key: string]: any;
-};
+}
 
 export type ViewMode = 'story' | 'docs';
-export type StoryContextForLoaders<
-  TRenderer extends Renderer = Renderer,
-  TArgs = Args
-> = StoryContextForEnhancers<TRenderer, TArgs> &
-  Required<StoryContextUpdate<TArgs>> & {
-    hooks: unknown;
-    viewMode: ViewMode;
-    originalStoryFn: StoryFn<TRenderer>;
-  };
+export interface StoryContextForLoaders<TRenderer extends Renderer = Renderer, TArgs = Args>
+  extends StoryContextForEnhancers<TRenderer, TArgs>,
+    Required<StoryContextUpdate<TArgs>> {
+  hooks: unknown;
+  viewMode: ViewMode;
+  originalStoryFn: StoryFn<TRenderer>;
+}
 
 export type LoaderFunction<TRenderer extends Renderer = Renderer, TArgs = Args> = (
   context: StoryContextForLoaders<TRenderer, TArgs>
-) => Promise<Record<string, any>>;
+) => Promise<Record<string, any> | void> | Record<string, any> | void;
```

Comment:
every loader function is a promise passed to Promise.all, so shouldn't the type be `Promise<Record<string, any> | void>` only?

Reply:
Promise.all also accepts non-promises:
<img width="340" alt="image" src="https://github.com/ComponentDriven/csf/assets/1035299/87b52dbd-298c-4e53-851e-ebec2869f218">

I think this is nice, as not everything that needs to happen before rendering is necessarily async.

Reply:
sure thing, but this will affect every place loaders are ever used:
- applyLoaders function
- defining loaders in user's Preview
- portable stories


If you think that's a good idea, then that's good!

Reply:
Yes, I think it is a good idea for all those cases :)